src/execution/sema/sema_checking.cpp

```diff
@@ -226,6 +226,12 @@ Sema::CheckResult Sema::CheckComparisonOperands(parsing::Token::Type op, const S
     return {result_type, left, right};
   }
 
+  // If both input types are integers, we don't need to do any work.
```

Comment:
Hey @pmenon, could you quickly weigh in on this change?

left: MemberExpr, it is queryState.some_uint32
right: LitExpr, integer literal in particular, treated as int32

I'm not sure if:
- there is a cast for narrowing/widening integers
- there is a more fine-grained way of comparing types here than IsIntegerType which should be used

Thanks!

Reply:
There isn't a means to down- or up-cast primitive integers at the moment. For now, I think what you have is okay. But, put a `TODO` putting the blame on me and revisiting with a fix. Also, create an issue for it.

Reply:
#1174. Thanks!